crates/gen-host-js/src/lib.rs

```diff
@@ -745,13 +814,33 @@ impl Instantiator<'_> {
 
         let index = import.index.as_u32();
         let callee = format!("lowering{index}Callee");
-        uwriteln!(
+        let import_specifier = self.gen.map_import(import_name);
+        if self.gen.opts.instantiation {
+            uwriteln!(
+                self.src.js,
+                "const {callee} = imports{}.{};",
+                if is_js_identifier(&import_specifier) {
+                    format!(".{}", import_specifier)
+                } else {
+                    format!("[\"{}\"]", import_specifier)
+                },
+                func.name.to_lower_camel_case(),
+            );
+        } else {
+            uwriteln!(
+                self.src.js,
+                "
+                    import {{ {} as {callee} }} from '{}';\
+                ",
+                func.name.to_lower_camel_case(),
+                self.gen.map_import(import_name),
+            );
+        }
+        uwrite!(
             self.src.js,
-            "const {callee} = imports.{}.{};",
-            import_name.to_lower_camel_case(),
-            func.name.to_lower_camel_case(),
+            "
+            function lowering{index}"
```

Comment:
The pattern I'm using for all intrinsics and top-level definitions since the formatting PR to ensure consistent double line spacing is to have a newline both before and after when they should be double spaced between eachother (and that's why there's the strip leading newline behaviour in the final write). Otherwise double spacing after each item leads to double spacing between items that intend to be single-spaced. I'm open to better layout suggestions though for sure! Updated to use `\n` for now.